lib/libzfs_core/libzfs_core.c

```diff
@@ -613,6 +613,9 @@ recv_impl(const char *snapname, nvlist_t *props, const char *origin,
 		if (props != NULL)
 			fnvlist_add_nvlist(innvl, "props", props);
```

Comment:
You might rename the argument to `recvdprops` to distinguish it from `localprops`.  (Probably not worth trying to change the nvlist key though, for backwards-compatibility with older kernels.)